### PR TITLE
Redundant unions

### PR DESCRIPTION
**Pull request by twoertwein**

### CODE REVIEW

File: pandas-stubs/_typing.pyi [hunk 1]
```diff
@@ -135,8 +135,12 @@ Scalar = Union[
     str,
     bytes,
     datetime.date,
+    datetime.datetime,
     datetime.timedelta,
+    bool,
     complex,
```
Comment (Dr-Irv) [inline pandas-stubs/_typing.pyi:141]: Can we bring back `int` and `float` here as well?

Comment (twoertwein) [reply pandas-stubs/_typing.pyi:143]: Added

Comment (twoertwein) [reply pandas-stubs/_typing.pyi:143]: added ignore for

> pandas-stubs/core/groupby/generic.pyi:145: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]

that was hidden by https://github.com/python/mypy/issues/13775

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein

**Pull request merged**